### PR TITLE
weather: cleanups, enable on all Unix platforms

### PR DESCRIPTION
**Pull request by danieldk**
###### Motivation for this change

* Run `nixpkgs-fmt` to fix non-standard indentation.
* Simplify `installPhase`.
* Enable on all Unix platforms.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

cc: @matthiasbeyer

### CODE REVIEW

**Review by matthiasbeyer — APPROVED** [review]
Diff LGTM

File: pkgs/applications/misc/weather/default.nix [hunk 1]
```diff
@@ -1,41 +1,46 @@
...
+
+  nativeBuildInputs = [
+    installShellFiles
+    pythonPackages.wrapPython
+  ];
+
+  dontConfigure = true;
+  dontBuild = true;
+
+  installPhase = ''
```
Comment (doronbehar) [inline pkgs/applications/misc/weather/default.nix:20]:
```suggestion
  # Upstream doesn't provide a setup.py or a like, so we follow:
  # http://fungi.yuggoth.org/weather/doc/install.rst#id3
  installPhase = ''
```

Comment (danieldk) [reply pkgs/applications/misc/weather/default.nix:22]: Good idea, added.

**Review by doronbehar — APPROVED** [review]

**Comment by doronbehar** [comment]
Diff LGTM.

**Comment by doronbehar** [comment]
Result of `nixpkgs-review pr 98442` [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>weather</li>
  </ul>
</details>

**Pull request merged**